File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int length = commands.length;
        int[] answer = new int[length];
        for (int i = 0; i < length; i++) {
            int[] command = commands[i];
            int[] copied = Arrays.copyOfRange(array, command[0] - 1, command[1]);
            Arrays.sort(copied);
            answer[i] = copied[command[2] - 1];
        }
        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> subVector(vector<int>& array, int i, int j) {
    return vector<int>(array.begin() + i, array.begin() + j);
}

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    for (auto& c : commands) {
        vector<int> temp = subVector(array, c[0] - 1, c[1]);
        sort(temp.begin(), temp.end());
        answer.push_back(temp[c[2] - 1]);
    }
    return answer;
}
